cluster/core/src/main/scala/com/lightbend/lagom/internal/cluster/JoinClusterImpl.scala

```diff
@@ -31,8 +35,21 @@ private[lagom] object JoinClusterImpl {
         "conflict with Akka Cluster Bootstrap or cause split-brain clusters.")
     }
 
-    if (cluster.settings.SeedNodes.isEmpty && joinSelf)
+    if (clusterBootstrapEnabled && joinSelf) {
+      throw new IllegalArgumentException(
+        "Both \"lagom.cluster.bootstrap.enabled\" and \"lagom.cluster.join-self\" are enabled, you should enable only one. " +
+          "Typically, \"lagom.cluster.bootstrap.enabled\" should be used in production while \"lagom.cluster.join-self\" in development and test environments"
+      )
+    }
+
+    // we give priority
```

Comment:
Incomplete sentence?

Reply:
Yes, incomplete. But also obsolete. 

I was planning to explain why we give precedence to cluster bootstrap. Since self-join is probably a mistake in Prod, if cluster bootstrap is enabled we should go for it. 

But in the meantime, we change the code and now we throw an exception when both (cluster bootstrap and self-join) are enabled.